### PR TITLE
Add "GSoC 2024: Half-precision in LLVM libc" blog post

### PR DESCRIPTION
**Pull request by overmighty**

### CODE REVIEW

**Comment by overmighty** [comment]
cc @asl @lntue @jhuber6

**Comment by overmighty** [comment]
When I first tested with Hugo 0.133.1, I got errors due to usage of deprecated features:

```
~/projects/llvm-blog-www main ➜ hugo server -D
Watching for changes in /home/overmighty/projects/llvm-blog-www/{content,static,themes}
Watching for config changes in /home/overmighty/projects/llvm-blog-www/config.toml, /home/overmighty/projects/llvm-blog-www/themes/hugo-kiera/config.toml
Start building sites …
hugo v0.133.1+extended linux/amd64 BuildDate=unknown

WARN  deprecated: .Site.Author was deprecated in Hugo v0.124.0 and will be removed in a future release. Use taxonomies instead.
ERROR deprecated: .Site.GoogleAnalytics was deprecated in Hugo v0.120.0 and will be removed in Hugo 0.134.0. Use .Site.Config.Services.GoogleAnalytics.ID instead.
ERROR deprecated: .Site.DisqusShortname was deprecated in Hugo v0.120.0 and will be removed in Hugo 0.134.0. Use .Site.Config.Services.Disqus.Shortname instead.
Built in 2326 ms
Error: error building site: logged 2 error(s)
~/projects/llvm-blog-www main ➜
```

I fixed them by replacing themes/hugo-kiera/ with the latest revision of https://github.com/funkydan2/hugo-kiera. The warning remains, I think it comes from the root config.toml.

But then the author doesn't render on any blog post.

It works correctly with Hugo 0.123.7.

**Review by lntue — APPROVED** [review]
The content LGTM to me!

I don't know much about the warnings/errors from the infra though.

File: content/posts/2024-08-31-half-precision-in-llvm-libc.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+   functions can be used with various compilers and architectures.
+2. Add generic implementations of `_Float16` basic operations for supported
+   architectures.
+3. Add optimized implementations of `_Float16` basic operations for specific
+   architectures using special hardware instructions and compiler builtins
+   whenever possible.
+4. Add generic implementations of as many `_Float16` higher math functions as
+   possible. We knew we would not have enough time to implement all of them.
+
+## Work done
```
Comment (asl) [inline content/posts/2024-08-31-half-precision-in-llvm-libc.md:33]: It would be great if some code references were added. E.g. links to commits, PRs, etc.

**Pull request merged**